### PR TITLE
pgraph.c: Clear Surfaces and Shader Cache

### PR DESCRIPTION
**Pull request by MasonT8198**
This should finish the pgraph_destroy function and clear out the surface and shader cache properly when pgraph_destroy is called.

### CODE REVIEW

File: hw/xbox/nv2a/pgraph.c [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/*
+sudo apt update/*
```
Comment (GXTX) [inline hw/xbox/nv2a/pgraph.c]: ?

Comment (MasonT8198) [reply hw/xbox/nv2a/pgraph.c]: Hmm, I pasted in two places at the same time apparently. 2 am programming tends to extend horrible results

**Comment by MasonT8198** [comment]
There may be a pointer that is being used incorrectly but it should be correct.

File: hw/xbox/nv2a/pgraph.c [hunk 2]
```diff
@@ -3550,12 +3550,13 @@ void pgraph_destroy(PGRAPHState *pg)
...
-    // TODO: clear out surfaces
-
+    // Clear out surfaces
+    lru_flush(&pg->surfaces);
     glDeleteFramebuffers(1, &pg->gl_framebuffer);
 
-    // TODO: clear out shader cached
-
+    // Clear out shader cache
+    lru_flush(&pg->shader_cache);
```
Comment (mborgerson) [inline hw/xbox/nv2a/pgraph.c]: shader_cache is not LRU type

File: hw/xbox/nv2a/pgraph.c [hunk 3]
```diff
@@ -3550,12 +3550,13 @@ void pgraph_destroy(PGRAPHState *pg)
 
     glo_set_current(g_nv2a_context_render);
 
-    // TODO: clear out surfaces
-
+    // Clear out surfaces
+    lru_flush(&pg->surfaces);
```
Comment (mborgerson) [inline hw/xbox/nv2a/pgraph.c:3554]: surfaces is not lru type

**Review by mborgerson — CHANGES_REQUESTED** [review]
See mentioned

File: hw/xbox/nv2a/pgraph.c (hunk 2 above)
Comment (MasonT8198) [reply hw/xbox/nv2a/pgraph.c]: Excuse my ineptitude but what type is shader_cache

File: hw/xbox/nv2a/pgraph.c (hunk 3 above)
Comment (MasonT8198) [reply hw/xbox/nv2a/pgraph.c:3554]: What type is surfaces

Googling didn't seem to get me anywhere meaningful

File: hw/xbox/nv2a/pgraph.c (hunk 2 above)
Comment (GXTX) [reply hw/xbox/nv2a/pgraph.c]: What type is `pg`? And you'll find your answer.

**Comment by mborgerson** [comment]
Let's return to this with a follow-up PR

**Pull request closed without merging**